### PR TITLE
add additional example for useDerivedValue

### PR DESCRIPTION
**Pull request by rawatnaresh**
## Description
Added example to run animation based on the props passed to the component. I guess this is a possible use case of `useDerivedValue`

### CODE REVIEW

**Comment by terrysahaidak** [comment]
Did you try to pass the prop to useAnimatedStyle directly?

And based on that prop return withTiming for the width?

File: docs/docs/api/useDerivedValue.md [hunk 1]
```diff
@@ -46,3 +46,49 @@ function App() {
...
+function ComponentA() {
+  const progress = useSharedValue(0);
+  const width = useDerivedValue(() => {
+    return progress.value * 250;
+  });
+
+  return (
+    <View>
+      <ProgressBar width={width} />
+      <Button onPress={() => (progress.value = Math.random())} />
```
Comment (karol-bisztyga) [inline docs/docs/api/useDerivedValue.md:69]: title prop is missing

**Comment by karol-bisztyga** [comment]
This works alright without `useDerivedValue`, plus the code you provided is not correct

```

import React from 'react';
import { Button, View, Text } from 'react-native';
import Animated, {
  useSharedValue,
  useAnimatedStyle,
} from 'react-native-reanimated';


function ComponentA() {
  const progress = useSharedValue(0);
  return (
    <View style={{ marginTop: 300 }}>
      <Button title="press" onPress={() => (progress.value = Math.random())} />
      <ProgressBar width={progress} />
    </View>
  );
}

function ProgressBar({ width }) {
  const widthStyle = useAnimatedStyle(() => {
    // if shared value was passed instead of derived shared value,
    // this worklet wouldn't have noticed the changes thus, no update would have happened
    return {
      width: width.value * 250,
    };
  });
  return (
    <Animated.View style={[widthStyle]}>
      <Text>I'll animate when you change width in ComponentA</Text>
    </Animated.View>
  );
}

export default ComponentA;
```

**Comment by jakub-gonet** [comment]
@rawatnaresh, what's the status on this?

**Comment by rawatnaresh** [comment]
@jakub-gonet, @karol-bisztyga is right. it works without `useDerivedValue`. 
initially, I tried it a couple of times without `useDerivedValue` and it didn't work so i had to use the another approach. But later I cleaned the build and installed the app again and it worked fine without `useDerivedValue`.
Thank you @karol-bisztyga for clarification 🙏

**Comment by jakub-gonet** [comment]
I'll close it then. Nevertheless, thanks for your effort, really appreciate it.

**Pull request closed without merging**